src/misc/tx/signAndBroadcastTransaction.ts

```diff
@@ -0,0 +1,206 @@
+      'value.content.value.plan.height',
+      new Long(get(transformedMsg, 'value.content.value.plan.height', 0)),
+    )
+    set(
+      transformedMsg,
+      'value.content.value',
+      Uint8Array.from(
+        SoftwareUpgradeProposal.encode(
+          SoftwareUpgradeProposal.fromPartial(get(transformedMsg, 'value.content.value')),
+        ).finish(),
+      ),
+    )
+  } else if (
+    get(msg, 'value.content.typeUrl') === '/cosmos.distribution.v1beta1.CommunityPoolSpendProposal'
+  ) {
+    set(
+      transformedMsg,
+      'value.content.value',
+      Uint8Array.from(
+        CommunityPoolSpendProposal.encode(
+          CommunityPoolSpendProposal.fromPartial(get(msg, 'value.content.value')),
+        ).finish(),
+      ),
+    )
+  }
+
+  return transformedMsg
+}
+
+const signAndBroadcastCosmosTransaction = async (
+  mnemonic: string,
+  crypto: string,
+  account: number,
+  change: number,
+  index: number,
+  transactionData: any,
+  ledgerTransport?: any,
+  onSignEnd?: () => void,
+): Promise<any> => {
+  const signerOptions = {
+    hdPaths: [
+      stringToPath(
+        `m/44'/${cryptocurrencies[crypto].coinType}'/${account || 0}'/${change || 0}/${index || 0}`,
+      ),
+    ],
+    prefix: cryptocurrencies[crypto].prefix,
+  }
+  let signer
+  if (!ledgerTransport) {
+    signer = await DirectSecp256k1HdWallet.fromMnemonic(mnemonic, signerOptions)
+  } else {
+    signer = new LedgerSigner(ledgerTransport, {
+      ...signerOptions,
+      ledgerAppName: cryptocurrencies[crypto].ledgerAppName,
+    } as any)
+  }
+  const accounts = await signer.getAccounts()
+  const client = await SigningStargateClient.connectWithSigner(
+    cryptocurrencies[crypto].rpcApiUrl,
+    signer,
+    {
+      registry,
+      aminoTypes: new AminoTypes({ additions: aminoAdditions, prefix: signerOptions.prefix }),
+    },
+  )
+  const tx = await client.sign(
+    accounts[0].address,
+    transactionData.msgs.map((msg: any) => formatTransactionMsg(msg)),
+    transactionData.fee,
+    transactionData.memo,
+  )
+  if (onSignEnd) {
+    onSignEnd()
+  }
+  const result = await client.broadcastTx(TxRaw.encode(tx).finish())
+  if (!result.rawLog.match(/^\[/)) {
+    throw new Error(result.rawLog)
+  }
+  return result
+}
+
+const signAndBroadcastTransaction = async (
+  // obtain mnemonic using decryptWallet from the useDecryptWallet hook
+  mnemonic: string,
+  password: string,
+  account: Account,
+  transactionData: any,
+  securityPassword: string,
+  ledgerTransport?: any,
+  onSignEnd?: () => void,
+): Promise<any> => {
+  const channel = new BroadcastChannel('forbole-x')
+  try {
+    // legacy, marked for deletion
+    // const { mnemonic } = await sendMsgToChromeExt({
+    //   event: 'viewMnemonicPhrase',
+    //   data: { password, id: account.walletId, securityPassword },
+    // })
+    // TODO: handle other ecosystem
+    const result = await signAndBroadcastCosmosTransaction(
+      mnemonic,
+      account.crypto,
```

Comment:
we have changed the `Account` interface so it should be `account.chain`, `account.hdPath.account`, `account.hdPath.change`, `account.hdPath.index`

Reply:
updated references